krm-functions/lib/interface/v1alpha1/interface_test.go

```diff
@@ -0,0 +1,440 @@
+/*
+ Copyright 2023 The Nephio Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var itface = `apiVersion: req.nephio.org/v1alpha1
+kind: Interface
+metadata:
+  name: n3
+  annotations:
+    config.kubernetes.io/local-config: "true"
+spec:
+  networkInstance:
+    name: vpc-ran
+  cniType: sriov
+  attachmentType: vlan
+`
+
+var itfaceEmpty = `apiVersion: req.nephio.org/v1alpha1
+kind: Interface
+metadata:
+  name: n3
+  annotations:
+    config.kubernetes.io/local-config: "true"
+`
+
+func TestNew(t *testing.T) {
```

Comment:
It maybe better to rename this as `TestNewFromYaml`

Reply:
changed